src/Behavior/MillisecondTimestamp.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace ipl\Orm\Behavior;
+
+use DateTime;
+use DateTimeZone;
+use ipl\Orm\Contract\PropertyBehavior;
+
+class MillisecondTimestamp extends PropertyBehavior
+{
+    public function fromDb($value, $key, $context)
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        $datetime = DateTime::createFromFormat('U.u', sprintf('%F', $value / 1000.0));
+        $datetime->setTimezone(new DateTimeZone(date_default_timezone_get()));
+
+        return $datetime;
+    }
+
+    public function toDb($value, $key, $context)
```

Comment:
I don't like that you normalize every input first to `DateTime`.

* If the input is numeric, it's interpreted as timestamp. (Timezone irrelevant)
* If the input is string, it's transformed to a timestamp. (Timezone is handled by `strtotime`)
* If the input is datetime, it can easily transformed to a timestamp. (Timezone, if any, is handled)

And since the database expects a timestamp, there's no need to handle `DateTime` objects except in the last case.